ambry-api/src/main/java/com/github/ambry/config/QuotaConfig.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.config;
+
+import com.github.ambry.quota.QuotaMode;
+import com.github.ambry.utils.Utils;
+import java.util.HashSet;
+import java.util.Set;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+/**
+ * Config for quota of various resources in Ambry.
+ */
+public class QuotaConfig {
+  public static final String QUOTA_ENFORCER_SOURCE_PAIR_INFO_STR = "quotaEnforcerSourcePairInfo";
+  public static final String ENFORCER_STR = "enforcer";
+  public static final String SOURCE_STR = "source";
+  public static final String QUOTA_CONFIG_PREFIX = "quota.";
+  public static final String REQUEST_QUOTA_THROTTLING_ENABLED = QUOTA_CONFIG_PREFIX + "request.throttling.enabled";
+  public static final String HOST_QUOTA_THROTTLING_ENABLED = QUOTA_CONFIG_PREFIX + "host.throttling.enabled";
+  public static final String QUOTA_THROTTLING_MODE = QUOTA_CONFIG_PREFIX + "throttling.mode";
+  public static final String REQUEST_QUOTA_ENFORCER_SOURCE_PAIR_INFO_JSON =
+      QUOTA_CONFIG_PREFIX + "request.enforcer.source.pair.info.json";
+  public static final String HOST_QUOTA_ENFORCER_FACTORIES = QUOTA_CONFIG_PREFIX + "host.enforcer.factories";
+  public static final String QUOTA_MANAGER_FACTORY_CLASS = QUOTA_CONFIG_PREFIX + "manger.factory.class";
+  public static final String DEFAULT_QUOTA_MANAGER_FACTORY_CLASS = "com.github.ambry.quota.AmbryQuotaManagerFactory";
+  public static final String DEFAULT_QUOTA_THROTTLING_MODE = QuotaMode.TRACKING.name();
+  public StorageQuotaConfig storageQuotaConfig;
+
+  /**
+   * Config to enable throttling on customer's account or container.
+   */
+  @Config(REQUEST_QUOTA_THROTTLING_ENABLED)
+  @Default("false")
+  public boolean requestQuotaThrottlingEnabled;
+
+  /**
+   * Config to enable throttling for host's hardware resources quota.
+   */
+  @Config(HOST_QUOTA_THROTTLING_ENABLED)
+  @Default("false")
+  public boolean hostQuotaThrottlingEnabled;
+
+  /**
+   * Serialized json containing pairs of enforcer classes and corresponding source classes.
+   * This information should be of the following form:
+   * <pre>
+   * {
+   *   "quotaEnforcerSourcePairInfo" : [
+   *     {
+   *       "enforcer":"com.github.ambry.quota.QuotaEnforcer",
+   *       "source": "com.github.ambry.quota.QuotaSource"
+   *     },
+   *     {
+   *       "enforcer":"com.github.ambry.quota.AnotherQuotaEnforcer",
+   *       "source": "com.github.ambry.quota.AnotherQuotaSource"
+   *     }
+   *   ]
+   * }
+   * </pre>
+   */
+  @Config(REQUEST_QUOTA_ENFORCER_SOURCE_PAIR_INFO_JSON)
+  public String requestQuotaEnforcerSourcePairInfoJson;
+
+  /**
+   * {@link Set} of host quota enforcers. Specified as comma separated list of enforcer classes.
+   */
+  @Config(HOST_QUOTA_ENFORCER_FACTORIES)
```

Comment:
I don't think we should separate host level throttler and request quota throttler. We can unify them with the same interface and just only one json configuration to include all of them.

Reply:
done.